gridstatus/ercot.py

```diff
@@ -576,6 +580,123 @@ def get_spp(
             )
         return Ercot._finalize_spp_df(df, settlement_point_field, locations)
 
+    @support_date_range(frequency="1Y", update_dates=ercot_update_dates)
+    def get_as_prices(
+        self,
+        date,
+        end=None,
+        verbose=False,
+    ):
+        """Get ancillary service clearing prices in hourly intervals in Day Ahead Market
+
+        Arguments:
+            date (datetime.date, str): date of delivery for AS services
+
+            end (datetime.date, str, optional): if declared, function will return
+                data as a range, from "date" to "end"
+
+            verbose (bool, optional): print verbose output. Defaults to False.
+
+        Returns:
+
+            pandas.DataFrame: A DataFrame with prices for "Non-Spinning Reserves", \
+                "Regulation Up", "Regulation Down", "Responsive Reserves".
+
+        Source:
+            https://www.ercot.com/mp/data-products/data-product-details?id=NP4-181-ER
+        """
+
+        # use to check if we need to pull daily files
+        if (
+            not end
```

Comment:
why this `not end`? what if I request data from last 3 days? does this work?

Reply:
it actually works whether it's `not end` or if it's `end`. It should be changed to check if end is declared, and then pass it to `_get_as_prices_recent()` as the default. If you don't check for `end` and you pass it as an argument, it gets picked up in the `args_dict` and throws an `AttributeError` when trying to convert `None` to a `timestamp`. I'll make the appropriate edits.

Reply:
updated decorator logic to delete `end` from `args_dict` if value is `None`. Allowed for simplification of above issue by removing the intermediate check on the value of `end`.